### ISSUE TITLE
Files already in library shouldn't be uploaded over the network

### ISSUE THREAD

**Issue by abshkbh** [gphotosuploader/gphotos-uploader-cli#195]
If I have a file A.jpg in my Google Photos Library.

If I have the exact same A.jpg on my Desktop and I try to upload it via cli it should be a fast operation. However, the cli's worker threads still try to upload the photos.

Try this on the web uploader, it's a no-op, it just modifies album metadata and doesn't upload anything.

Hoping there is an API to check this, think rsync.

**Comment by pacoorozco** [comment]
Hi @abshkbh 

If you try to upload a file that is already present in your Google Photos repository, the file will not be created again. [Google Photos prevents this duplication](https://developers.google.com/photos/library/guides/upload-media#uploading-bytes).

But, the file will be uploaded anyways, because there is no way to know if the file was already uploaded to your Google Photos account. **There is no API call to do it**. 

The above is true for **generic** files, but `gphotos-uploader-cli` tries to reduce the number of upload calls, so **it keeps the already uploaded files database locally**. Before starting any upload will try to guess if the file was already uploaded ([here](https://github.com/gphotosuploader/gphotos-uploader-cli/blob/master/datastore/completeduploads/completeduploads.go)). It uses a hash of the file and the last modified date to do it.

So, in summary, we track which files `gphotos-uploader-cli` has already uploaded, to not uploaded again. If we upload a file, that was already on Google Photos, the service will not create it again. 

I hope this can answer your issue, if not just let me know :-D
